### PR TITLE
Feature/plcn 267 move reusable code from hubspot to application

### PR DESCRIPTION
**Pull request by Nicolai-Bak**

### CODE REVIEW

File: test/Pelican.Infrastructure.HubSpot.Test/Services/HubSpotClientServicesTests.cs [hunk 1]
```diff
@@ -22,63 +23,33 @@ public HubSpotClientServicesTests()
 	}
 
 	[Fact]
-	public async Task GetByIdAsync_ClientReturnsNull_ReturnFailure()
+	public void HubSpotClientService_ClientNull_ThrowException()
```
Comment (AntonEsbjergIT-Minds) [inline test/Pelican.Infrastructure.HubSpot.Test/Services/HubSpotClientServicesTests.cs:26]: This test should also be added to HubSpotAccountManagerServicesTests

File: test/Pelican.Infrastructure.HubSpot.Test/Services/HubSpotContactServicesTests.cs [hunk 2]
```diff
@@ -21,63 +21,21 @@ public HubSpotContactServicesTests()
 		_uut = new HubSpotContactService(_hubSpotClientMock.Object);
```
Comment (AntonEsbjergIT-Minds) [inline test/Pelican.Infrastructure.HubSpot.Test/Services/HubSpotContactServicesTests.cs:21]: Maybe also add test to see what happens if a null value is given as hubSpotClient

File: test/Pelican.Infrastructure.HubSpot.Test/Services/HubSpotDealServicesTests.cs [hunk 3]
```diff
@@ -21,63 +24,21 @@ public HubSpotDealServicesTests()
 		_uut = new HubSpotDealService(_hubSpotClientMock.Object);
```
Comment (AntonEsbjergIT-Minds) [inline test/Pelican.Infrastructure.HubSpot.Test/Services/HubSpotDealServicesTests.cs:24]: Maybe add a test for what happens if null is given as Hubspotclient

**Review by AntonEsbjergIT-Minds — CHANGES_REQUESTED** [review]

File: src/Pelican.Domain/Settings/HubSpotSettings.cs [hunk 4]
```diff
@@ -1,8 +1,7 @@
 ﻿namespace Pelican.Domain.Settings;
 
-public sealed class HubSpotSettings
+public sealed class HubSpotSettings : BaseSettings
```
[the remark this answers is not in the document]
Comment (Nicolai-Bak) [reply src/Pelican.Domain/Settings/HubSpotSettings.cs]: Made base class to abstract RestSharp implementation. 
Now RestSharp implementation can be used with or without baseurl with the same implementation

**Review by AntonEsbjergIT-Minds — APPROVED** [review]
